mozilla/addons-frontend
# rm sponsored and verified categories

Fixes mozilla/addons#15038

Removes references to Verified and Sponsored promoted groups.  They've never been used in production but we (confusingly) offer them as filtering choices, and supported badging for them, etc.

### Testing
tl;dr make sure nothing is broken by removing obsolete code
- hero area (homepage) still correctly labels add-ons are By Firefox or Recommended
- Search/category pages don't offer verified or sponsored as choices in badging select
- badging filter works as expected
- add-ons that are in recommended or By Firefox promoted groups get a badge on their listing page
